New package: TwPrototypes v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: TwPrototypes
- Repository: https://github.com/bgctw/TwPrototypes.jl
- Created by: @bgctw
- Version: v0.1.0
- Commit: a7cb63ce68aafef91e612052d5b0e8330b1f7613
- Git reference: HEAD

bgctw commented:
This package is used in several science projects and referenced in the respective Project.toml. In order to support Open Science, the package needs to be accessible by the broad community and therefore to be registered at General.

JuliaHubRegistrator committed:
New package: TwPrototypes v0.1.0

UUID: 9ae22f58-2487-4805-bfc5-386577db46c7
Repo: https://github.com/bgctw/TwPrototypes.jl.git
Tree: 3593f1f45dfb368d39c77d5e9311bd235d57a3b9

Registrator tree SHA: 7242ef64be5953fac9bf8b3efa3a7c4d2d44ca09

gdalle commented:
Hi @bgctw, and thank you for contributing to the ecosystem!
I am not sure this package really needs to be registered. Based on the docs, it mostly contains three small and very different utilities, which suggests that it is a collection of utilities for your projects, not necessarily for the community at large.
If you want to use it in your open science endeavors (which I support!), maybe a [local registry](https://github.com/GunnarFarneback/LocalRegistry.jl) would be an option? Alternately, you can try to contribute these utilities to the relevant packages and see what they think?

bgctw commented:
Thanks @gdalle for looking into it.

Yes, as the package description says, it is a collection of functionality that has not found its proper place/project yet. As I understood the suggested local registry, its intended for "local" work. However, for reproducible science, I would need a publicly accessible Registry. Do you know of such a registry hosted by an institution that can provide such packages that are not suitable for General?

gdalle commented:
I have never used LocalRegistry myself but I think you can host said registry on a public GitHub repository. Although it's probably more useful if you have several such packages to maintain together. For a single one, you can just keep it in a plain GitHub repo and add it with its URL to downstream projects. You will also need to check the Manifest.toml of these downstream projects into GitHub

bgctw commented:
Thanks again.
The suggestion of [adding it as unregistered public github package](https://pkgdocs.julialang.org/v1/managing-packages/#Adding-unregistered-packages) seems to fit my use case.  Hence, this package does not need to be registered with General.

gdalle commented:
Perfect! This PR will expire automatically without further action needed
